app/src/store/voice/accessToken.ts

```diff
@@ -91,18 +107,27 @@ export const accessTokenSlice = createSlice({
       return { status: 'fulfilled', value: action.payload };
     });
 
-    builder.addCase(getAccessToken.rejected, (_, action) => {
+    builder.addCase(getAccessToken.rejected, (state, action) => {
       switch (action.payload?.reason) {
         case 'USER_NOT_FULFILLED':
           return {
             status: 'rejected',
             reason: action.payload.reason,
+            error: action.payload.error,
           };
         case 'TOKEN_RESPONSE_NOT_OK':
+          state = {
```

Comment:
The way builder case functions work is that we either mutate the state or return a new state entirely. It seems like we're doing both here.

Reply:
oops old habit

Reply:
Hmm, we shouldn't be returning anything if we're mutating state. It's one or the other.

Reply:
Either we perform the state assignment on line 112, or the return statement on 118. We cannot do both.